dotCMS/src/main/java/com/dotmarketing/portlets/languagesmanager/business/LanguageFactoryImpl.java

```diff
@@ -629,8 +632,14 @@ protected int deleteLanguageById(final Language language) {
 			Logger.debug(this, ()-> "Deleting the language by id: " + id);
 			rowsAffected = new DotConnect().executeUpdate(DELETE_FROM_LANGUAGE_WHERE_ID, id);
 		} catch (DotDataException e) {
-			Logger.error(LanguageFactoryImpl.class, "deleteLanguageById failed to delete the language with id: " + id);
-			throw new DotRuntimeException(e.toString(), e);
+			if(e.getMessage().contains("fk_contentlet_version_info_lang")
```

Comment:
is it possible create and throw a new custom exception here?

Reply:
Would be risky since this is thrown from DotConnect